mesh/mesh.go

```diff
@@ -432,44 +431,44 @@ func (m *Mesh) AccumulateRewards(rewardLayer types.LayerID, params Config) {
 	}
 
 	ids := make([]types.Address, 0, len(l.Blocks()))
-	uq := make(map[types.Address]int)
-
-	// TODO: instead of the following code we need to validate the eligibility of each block individually using the
-	//  proof included in each block
 	for _, bl := range l.Blocks() {
+		valid, err := m.ContextualValidity(bl.ID())
```

Comment:
What happens if `bl.ID()` isn't valid? per @almogdepaz the db will return an error when queried on an unknown key